### PR TITLE
ROS_DEBUG prints incorrect gen_id & incorrect namespace for /latch_xy_goal_tolerance

### PR DESCRIPTION
**Pull request by veeraragav**
1)base_local_planner/simple_scored_sampling_planner.cpp:60 ROS_DEBUG prints incorrect gen_id. It skips gen_id++(line 74) whenever `if (score_function_p->getScale() == 0)` is true.
2)The parameter `latch_xy_goal_tolerance` in latched_stop_rotate_controller.cpp was in global namespace. Made corrections to make it `~<name>/latch_xy_goal_tolerance ` according to dwa_local_planner's wiki page.

### CODE REVIEW

File: base_local_planner/src/simple_scored_sampling_planner.cpp [hunk 1]
```diff
@@ -53,6 +53,7 @@ namespace base_local_planner {
     for(std::vector<TrajectoryCostFunction*>::iterator score_function = critics_.begin(); score_function != critics_.end(); ++score_function) {
       TrajectoryCostFunction* score_function_p = *score_function;
       if (score_function_p->getScale() == 0) {
+	gen_id++;
```
Comment (DLu) [inline base_local_planner/src/simple_scored_sampling_planner.cpp]:
```suggestion
        gen_id++;
```

File: base_local_planner/src/latched_stop_rotate_controller.cpp [hunk 2]
```diff
@@ -19,12 +19,13 @@
...
-LatchedStopRotateController::LatchedStopRotateController(const std::string& name) {
-  ros::NodeHandle private_nh("~/" + name);
-  private_nh.param("latch_xy_goal_tolerance", latch_xy_goal_tolerance_, false);
+LatchedStopRotateController::LatchedStopRotateController() {}
 
-  rotating_to_goal_ = false;
-}
+void LatchedStopRotateController::initialize(const std::string& name) {
+    ros::NodeHandle private_nh("~/" + name);
+    private_nh.param("latch_xy_goal_tolerance", latch_xy_goal_tolerance_, false);
```
Comment (DLu) [inline base_local_planner/src/latched_stop_rotate_controller.cpp]: Please change this to search for the parameter as well, for people who relied on the old configuration.

**Review by DLu — CHANGES_REQUESTED** [review]

File: base_local_planner/src/latched_stop_rotate_controller.cpp (hunk 2 above)
Comment (veeraragav) [reply base_local_planner/src/latched_stop_rotate_controller.cpp]: May I know which parameter are you talking about?

Comment (veeraragav) [reply base_local_planner/src/latched_stop_rotate_controller.cpp]: You mean checking if `latch_xy_goal_tolerance` is set? Or search for the same in global namespace as well as in planner namespace? like parameter_magic.h in `nav_core` package?

Comment (DLu) [reply base_local_planner/src/latched_stop_rotate_controller.cpp]: No, just use the standard searchParam functionality to check the `~/name` namespace as well as `~/`

Comment (veeraragav) [reply base_local_planner/src/latched_stop_rotate_controller.cpp]: Got it. Will do.

Comment (veeraragav) [reply base_local_planner/src/latched_stop_rotate_controller.cpp]: Did the change you had asked for.

**Review by veeraragav — COMMENTED** [review]
@DLu As per your request, now `latch_xy_goal_tolerance` will be checked in global namespace as well.

**Comment by veeraragav** [comment]
Shall I create pull request for other branches as well ?

**Pull request merged**